### ISSUE TITLE
LoggerEmailUtils_Tests failed

### ISSUE THREAD

**Issue by Firtzberg** [jongpie/NebulaLogger#276]
### Package Edition of Nebula Logger

Unlocked Package

### Package Version of Nebula Logger

https://github.com/jongpie/NebulaLogger/releases/tag/v4.6.16

### New Bug Summary

After a refresh of the full and partial sandbox two tests are failing.

```
System.EmailException: SendEmail failed. First exception on row 0; first error: NO_MASS_MAIL_PERMISSION, Single email is not enabled for your organization or profile.: []
--
Stack trace: | Class.LoggerEmailUtils.sendEmail: line 111, column 1 Class.LoggerEmailUtils.sendErrorEmail: line 81, column 1 Class.LoggerEmailUtils.sendErrorEmail: line 63, column 1 Class.LoggerEmailUtils_Tests.it_should_send_email_notification_for_upsertResult_errors_when_enabled: line 63, column 1
```

```
System.EmailException: SendEmail failed. First exception on row 0; first error: NO_MASS_MAIL_PERMISSION, Single email is not enabled for your organization or profile.: []
--
Stack trace: | Class.LoggerEmailUtils.sendEmail: line 111, column 1 Class.LoggerEmailUtils.sendErrorEmail: line 81, column 1 Class.LoggerEmailUtils.sendErrorEmail: line 47, column 1 Class.LoggerEmailUtils_Tests.it_should_send_email_notification_for_saveResult_errors_when_enabled: line 16, column 1
```

**Comment by jongpie** [comment]
@Firtzberg thanks for reporting this! This should be a fairly quick fix - I'll include a bugfix for this in the next release, `v4.7.0` (PR #274). The next release will be out in about 2 weeks (once all orgs have been upgraded to the Spring '22 release)

**Comment by jongpie** [comment]
@Firtzberg this should be fixed in the new release [v4.7.0](https://github.com/jongpie/NebulaLogger/releases/tag/v4.7.0).

**Comment by gjslagle12** [comment]
@jongpie After updating to v4.7.0, I still received errors when running unit tests on a sandbox org with email deliverability turned off.   

**Unit test error message:**
```
LoggerEmailUtils_Tests.it_should_send_email_notification_for_saveResult_errors_when_enabled
Message: System.AssertException: Assertion Failed: Email should have been sent: Expected: 1, Actual: 0

LoggerEmailUtils_Tests.it_should_send_email_notification_for_upsertResult_errors_when_enabled
Message: System.AssertException: Assertion Failed: Email should have been sent: Expected: 1, Actual: 0
```




I see the fix in the LoggerEmailUtils class, but in the corresponding LoggerEmailUtils_Test class, it doesn't look like the logic was updated to use the fix.  So, the assertions (on lines 18 and 65) only ever reflect "1", and not "0".  


Wondering if [something like] this would work on the LoggerEmailUtils_Test class (in those two locations):
```
   if (LoggerEmailUtils.IS_EMAIL_DELIVERABILITY_ENABLED()) 
        {System.assertEquals(1, Limits.getEmailInvocations(), 'Email should have been sent');}
        {System.assertEquals(0, Limits.getEmailInvocations(), 'Deliverability is turned off');}
      )
```

(pardon the code - not a programmer)

**Comment by jongpie** [comment]
Hi @gjslagle12, that's embarrassing that I didn't fully fix it, thanks for letting me know that it's still causing issues! I'll get the test class fixed - I have a release coming out soon in PR #298 (which will be `v4.7.1`), I'll include this fix in that PR. 

And just FYI, in the new release, the class is being renamed from `LoggerEmailUtils`/`LoggerEmailUtils_Tests` to `LoggerEmailSender`/`LoggerEmailSender_Tests`.

**Comment by jongpie** [comment]
@gjslagle12 the new [`v4.7.1` release](https://github.com/jongpie/NebulaLogger/releases/tag/v4.7.1) was released yesterday, which should contain the fixes for those failing tests. Let me know if you encounter any more issues!

**Comment by gjslagle12** [comment]
@jongpie  Still getting errors on the same two lines during unit tests, but the error message has changed:

-----------------
Test: LoggerEmailSender_Tests.it_should_send_email_notification_for_saveResult_errors_when_enabled
Message: System.ListException: List index out of bounds: 0

Test: LoggerEmailSender_Tests.it_should_send_email_notification_for_upsertResult_errors_when_enabled
Message: System.ListException: List index out of bounds: 0

**Comment by jongpie** [comment]
@gjslagle12 that's double embarrassing that it's still broken! The last time that I tried to fix this, I unfortunately didn't have access to a sandbox to test in (and email deliverability can't be disabled in scratch orgs), so I didn't do as thorough testing as I should have. But, good news - I now have another sandbox that I can test it, and I was able to reproduce those test failures. I'll plan to re-re-fix this in the next release, `v4.7.3`.
